pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -66,27 +64,22 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 		}
 	}
 
-	if upgradeableUpgradeInProgress := resourcemerge.FindOperatorStatusCondition(cv.Status.Conditions, UpgradeInProgress); upgradeableUpgradeInProgress != nil && upgradeableUpgradeInProgress.Status == configv1.ConditionTrue {
+	if condition := resourcemerge.FindOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorProgressing); condition != nil &&
+		condition.Status == configv1.ConditionTrue &&
+		condition.Reason == ConditionReasonMinorVersionClusterUpgradeInProgress {
```

Comment:
But we don't need this condition, right?  5ea5ebe0e12bf1 is pointing out that we want to be `Upgradeable=False` and blocking 4.y.z' -> 4.(y+1).z" retargets regardless of whether the update we're currently in the middle of is 4.y.z -> 4.y.z' or 4.(y-1).z -> 4.y.z'.

Reply:
I thought we did not need to any change on the consumer side and `Progressing=True` in the conditions would just be taken care of by some existing code to make `Upgradable=False`.
The Reason check was for the non-blocking case for acceptedRisks.

Now, I recover the logic of generating `precondition.Error` with the latest patch.